File: autocomplete/models/Product.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Product struct {
	ID          primitive.ObjectID `json:"id" bson:"_id"`
	Name        string             `json:"name"`
	Description string             `json:"description"`
	Categories  Categories         `json:"categories"`
	Price       string             `json:"price"`
	ImageUrl    []string           `json:"imageUrl"`
}

type Products []Product

File: backend/models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
namespace Backend.Models;

public class Product
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? id { get; set; }
    public string? name { get; set; }
    public string? description { get; set; }
    public List<Category> categories { get; set; } = new List<Category>();
    public decimal price { get; set; }
    public List<string>? imageUrl { get; set; }
}